app/core/models.py

```diff
@@ -0,0 +1,89 @@
+import uuid
+import os
+from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, \
+    PermissionsMixin
+
+from django.conf import settings
+
+
+def recipe_image_file_path(instance, filename):
+    """Generate filepath for recipe image"""
+    ext = filename.split('.')[-1]
+    filename = f'{uuid.uuid4()}.{ext}'
+    return os.path.join('uploads/recipe/', filename)
```

Comment:
OK here, but using [Django's `reverse` function](https://docs.djangoproject.com/en/3.0/ref/urlresolvers/#reverse) would harden this part for changes of the URL

Reply:
Sorry i didnt quite understand.